corehq/apps/hqwebapp/two_factor_gateways.py

```diff
@@ -194,6 +213,13 @@ def get_ip_address():
 )
 
 
+def _report_usage(ip, num, user):
+    two_factor_rate_limiter_per_ip.report_usage('ip:{}'.format(ip))
+    two_factor_rate_limiter_per_user.report_usage('user:{}'.format(user))
+    two_factor_rate_limiter_per_number.report_usage('number:{}'.format(num))
+    global_two_factor_setup_rate_limiter.report_usage()
```

Comment:
Nit: since order doesn't matter and it's repetitive, it might be nice to put these in the same order as above (global, ip, number, user) and also to call `num` `number`. Both of those come from the principle that "Things that are the same should look the same" / "Things should only look different if they are different", and generally the principle of least surprise. It's admittedly minor

Reply:
Agree with order. Not sure if I'm able to change `num` due to the nature of this function being defined within another function. I noticed a complaint that the `number` declaration in params shadows name from outer scope. To be safe I was avoiding this, but if there is a better way I'm open to hearing it.

Reply:
Maybe I'm confused. This doesn't seem to be defined within another function right now (though maybe locally you originally had?).

I missed this originally but the same applies to `ip` => `ip_address`, `user` => `username`. In general if you do need to rename variables so they aren't reused, again according to the above principle try not to make them arbitrarily different and instead name them with a pattern based off the "original" name you're trying not to conflict with. So in this case you might use `ip_address_`, `number_`, `username_`, where the right-trailing `_` signifies "I had to change the variable name so as not to conflict".

Again, in this case though, it looks like you could just reuse the same variable names with no conflict

Reply:
Yes you are correct. I was looking at the wrong method. I fixed `_report_usage`